stablehlo/dialect/StablehloOps.td

```diff
@@ -3462,32 +3462,30 @@ def StableHLO_DynamicGatherOp: StableHLO_Op<"dynamic_gather",
       InferTensorTypeWithReify]> {
   let summary = "DynamicGather operation";
   let description = [{
-    This operation is a work in progress, so it is not yet included in
-    the StableHLO specification: https://github.com/openxla/stablehlo/issues/8.
-
-    Informally, this operation does the same thing as GatherOp except
-    that `slice_sizes` are specified dynamically:
-    https://github.com/openxla/stablehlo/blob/main/docs/spec.md#gather
+    This operation is functionally identical to
+    [iota](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#gather)
+    op, with the `slice_sizes` specified dynamically as an operand.
 
     Example:
     ```mlir
+    %slice_sizes = stablehlo.constant dense<[1, 2, 2]> : tensor<3xi32>
     %result = "stablehlo.dynamic_gather"(%operand, %start_indices, %slice_sizes) {
       dimension_numbers = #stablehlo.gather<
         offset_dims = [2, 3],
         collapsed_slice_dims = [0],
         start_index_map = [0, 2],
         index_vector_dim = 2>,
       indices_are_sorted = false
-    } : (tensor<3x4x2xi32>, tensor<2x3x2xi64>, tensor<3xi64>) -> tensor<2x3x2x2xi32>
+    } : (tensor<3x4x2xi32>, tensor<2x3x2xi64>, tensor<3xi32>) -> tensor<2x3x2x2xi32>
     ```
   }];
 
   let arguments = (ins
-    HLO_Tensor:$operand,
-    HLO_IntTensor:$start_indices,
-    HLO_IntTensor:$slice_sizes,
-    StableHLO_GatherDimensionNumbers:$dimension_numbers,
-    DefaultValuedOptionalAttr<BoolAttr, "false">:$indices_are_sorted
+    HLO_Tensor:$operand /*dynamic_gather_i1*/,
+    HLO_IntTensor:$start_indices /*dynamic_gather_i2*/,
+    HLO_StaticDimensionTensor:$slice_sizes /*dynamic_gather_i3*/,
```

Comment:
Why this change?

Reply:
Otherwise the slice_sizes can be dynamic, but we don't want to allow that flexibility because that is only useful for an unranked operand (if the operand is ranked, then we should know the size of slice_sizes).

Reply:
This will allow the ODS to enforce that slize_size is 1-D for dynamic_gather op. Only this new here is the inclusion of `Index` type, but we are already using `Index` for things other that axis (e.g.  dynamic_pad op)

Reply:
Let's not expand support for `index` I don't think? Leaves us in an inconsistent state.. I.e. why not for `start_indices`?

Reply:
Ah good point. Looks like we need `HLO_StaticIntTensor` :upside_down_face:

Reply:
Agreed. `Static1DIntTensor` to be precise.

Reply:
Let us address this is a follow up PR.

Reply:
https://github.com/openxla/stablehlo/pull/2320